js/reference.js

```diff
@@ -346,21 +346,82 @@ var ERMrest = (function(module) {
          * or errors (TBD).
          */
         create: function(data) {
+            var self = this;
             try {
-                // TODO
                 //  verify: data is not null, data has non empty tuple set
+                verify(data, "'data' must be specified");
+                verify(data.length > 0, "'data' must have at least one row to create");
+
+                var defer = module._q.defer();
+
                 //  get the defaults list for the referenced relation's table
-                //  get the data
+                var defaults = getDefaults();
```

Comment:
I think we need to meet and discuss this tomorrow with @hongsudt and @karlcz. Our UX doesn't have a way for the user to state what values need to be set by default and what ones they want to override. This becomes more complex when there are multiple entities being posted and you want `entity 1` to have `col A` set by default and want to define `col A` for `entity 2`.